### PR TITLE
Throw a more detailed error message when creating invalid frozen actors

### PR DESCRIPTION
**Pull request by abcdefg30**
Before it crashed with an exception like this:
```
Exception of type `System.ArgumentException`: Bounds of actor concreteb 38 (invalid) are empty.
Parametername: bounds
```

I think this error message is more helpful, as it immediately tells you that `FrozenUnderFog` is part of the problem, so that you can decide if you either remove that or fix the bounds of the actor.

### CODE REVIEW

**Review by pchote — COMMENTED** [review]
I think the best way to do this would be to implement `RulesetLoaded` on the `FrozenUnderFogInfo` and then throw a `YamlException` if the actor info doesn't include `SelectableInfo` or `IAutoSelectionSizeInfo` (which you'll need to define and attach to traitinfos that use `IAutoSelectionSize`).

This will make the linter catch the error instead waiting for game runtime.

**Comment by abcdefg30** [comment]
Updated.

File: OpenRA.Game/Traits/TraitsInterfaces.cs [hunk 1]
```diff
@@ -95,6 +95,8 @@ public Damage(int damage)
 	public interface ITick { void Tick(Actor self); }
 	public interface ITickRender { void TickRender(WorldRenderer wr, Actor self); }
 	public interface IRender { IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr); }
+
+	public interface IAutoSelctionSizeInfo : ITraitInfoInterface { }
```
Comment (rob-v) [inline OpenRA.Game/Traits/TraitsInterfaces.cs]: type name typo

**Comment by abcdefg30** [comment]
Fixed.

**Comment by pchote** [comment]
> OpenRA.Mods.Cnc/Traits/Render/WithVoxelWalkerBody.cs:L25: [SP2100] Code line must be no longer than 180 characters (now 181).

**Comment by abcdefg30** [comment]
Fixed, thanks.

File: OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs [hunk 2]
```diff
@@ -18,11 +19,17 @@
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("This actor will remain visible (but not updated visually) under fog, once discovered.")]
-	public class FrozenUnderFogInfo : ITraitInfo, Requires<BuildingInfo>, IDefaultVisibilityInfo
+	public class FrozenUnderFogInfo : ITraitInfo, Requires<BuildingInfo>, IDefaultVisibilityInfo, IRulesetLoaded
 	{
 		[Desc("Players with these stances can always see the actor.")]
 		public readonly Stance AlwaysVisibleStances = Stance.Ally;
 
+		void IRulesetLoaded<ActorInfo>.RulesetLoaded(Ruleset rules, ActorInfo info)
```
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs]: In all other places except one the interface isn't implemented explicitly (IRulesetLoaded<ActorInfo>...).

File: OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs [hunk 3]
```diff
@@ -18,11 +19,17 @@
...
+		{
+			if (!info.HasTraitInfo<SelectableInfo>() && !info.HasTraitInfo<IAutoSelectionSizeInfo>())
+				throw new InvalidOperationException("Cannot create a frozen actor for actor type '{0}' with empty bounds (no selection size given).".F(info.Name));
```
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs]: Should it be YamlException?

File: OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs (hunk 2 above)
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs]: IIRC the aim is to have all interfaces explicit sooner or later (see #10143), so I always try to make all interface implementations I touch/add explicit.

File: OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs (hunk 3 above)
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs]: Changed.

**Review by rob-v — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6ac619c5d3a873ac9f042208c9760b2e591c282c...65eafe881043cc624cf319314a02279b0e608b16

**Pull request merged**